File: go/src/ipaddress/prefix/prefix128/prefix128.go
package prefix128

// import "./prefix"
import "../../ip_bits"
import "fmt"
import "../../prefix"

///
///  Creates a new prefix object for 128 bits IPv6 addresses
///
///    prefix = IPAddressPrefix128.new 64
///      ///  64
///
func New(num uint8) prefix.ResultPrefix {
    if num <= 128 {
        //static _FROM: &'static (Fn(&Prefix, usize)(*Prefix, *string)) = &from;
        //static _TO_IP_STR: &'static (Fn(&Vec<u16>)String) = &Prefix128::to_ip_str;
        ipBits := ip_bits.V6()
        bits := ipBits.Bits;
        return &prefix.Ok{&prefix.Prefix {
            num,
            ipBits,
            *prefix.New_netmask(num, bits),
            From, // vt_to_ip_str: _TO_IP_STR
        }};
    }
    tmp :=  fmt.Sprintf("Prefix must be in range 0..128, got: %d", num);
    return &prefix.Error{&tmp}
}

func From(my *prefix.Prefix, num uint8) prefix.ResultPrefix {
    return New(num);
}

File: python/src/ip_address/prefix128.py
from typing import Optional

from .ip_bits import IpBits
from .prefix import Prefix


class Prefix128:
    # Creates a new prefix object for 128 bits IPv6 addresses
    #
    #   prefix = IPAddressPrefix128.new 64
    #     // => 64
    #
    @staticmethod
    def create(num: int) -> Optional[Prefix]:
        if num <= 128:
            ip_bits = IpBits.v6()
            bits = ip_bits.bits
            return Prefix(num, ip_bits, Prefix.new_netmask(num, bits), Prefix128.from_)
        return None

    @staticmethod
    def from_(my: Prefix, num: int) -> Optional[Prefix]:
        return Prefix128.create(num)


__all__ = ["Prefix128"]
